### ISSUE TITLE
Redundant parentheses in cast_lossless

### ISSUE THREAD

**Issue by dtolnay** [rust-lang-nursery/rust-clippy#2003]
```rust
fn main() {
    let _ = (1u8 + 2u8) as u64;
}
```

```
warning: casting u8 to u64 may become silently lossy if types change
 --> src/main.rs:2:13
  |
2 |     let _ = (1u8 + 2u8) as u64;
  |             ^^^^^^^^^^^^^^^^^^ help: try: `u64::from((1u8 + 2u8))`
  |
  = note: #[warn(cast_lossless)] on by default
```

I would expect it to suggest `u64::from(1u8 + 2u8)`.

@sunfishcode

**Comment by sunfishcode** [comment]
A reasonable expectation :-). I submitted https://github.com/rust-lang-nursery/rust-clippy/pull/2004.

**Comment by Manishearth** [comment]
fixed in #2004
